Client timeout if an image build steps takes more than 30 seconds

My build context is about 1 GB and it takes a minute or two to transfer. Also, I don't want it compressed, most of it is already compressed files. The 30 seconds timeout in DockerCilent is not enough to complete this build:

Caused by: com.spotify.docker.client.DockerTimeoutException: Timeout: POST http://localhost:4243/v1.12/build
        at com.spotify.docker.client.ProgressStream.hasNextMessage(ProgressStream.java:59)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:688)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:651)
...
Caused by: java.net.SocketTimeoutException: Read timed out
        at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:136)
        at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:152)
        at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:270)
        at org.apache.http.impl.io.ChunkedInputStream.getChunkSize(ChunkedInputStream.java:240)
        at org.apache.http.impl.io.ChunkedInputStream.nextChunk(ChunkedInputStream.java:206)
        at org.apache.http.impl.io.ChunkedInputStream.read(ChunkedInputStream.java:169)
        at org.apache.http.conn.EofSensorInputStream.read(EofSensorInputStream.java:137)
        at org.glassfish.jersey.message.internal.EntityInputStream.read(EntityInputStream.java:101)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream.read(ReaderInterceptorExecutor.java:300)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.core.json.UTF8StreamJsonParser.loadMore(UTF8StreamJsonParser.java:169)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.core.json.UTF8StreamJsonParser._skipWSOrEnd(UTF8StreamJsonParser.java:2495)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:617)
        at com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.MappingIterator.hasNextValue(MappingIterator.java:159)
        at com.spotify.docker.client.ProgressStream.hasNextMessage(ProgressStream.java:57)
        ... 56 more

I think build() should use noTimeoutResource() instead of resource().

--- a/README.md
+++ b/README.md
@@ -34,6 +34,12 @@ docker.killContainer(id);
 docker.removeContainer(id);
 ```
 
+### Configuration
+
+Both `DefaultDockerClient.builder()` and `DefaultDockerClient.fromEnv()` return a
+`DefaultDockerClient.Builder`. The builder can be used to configure and build clients with custom
+timeouts, connection pool sizes, and other parameters.
+
 ### Unix socket support
 
 Unix socket support is available on Linux since v2.5.0:

--- a/src/main/java/com/spotify/docker/client/DefaultDockerClient.java
+++ b/src/main/java/com/spotify/docker/client/DefaultDockerClient.java
@@ -979,6 +979,9 @@ public class DefaultDockerClient implements DockerClient, Closeable {
       return this;
     }
 
+    /**
+     * Set the URI for connections to Docker.
+     */
     public Builder uri(final String uri) {
       return uri(URI.create(uri));
     }
@@ -987,6 +990,10 @@ public class DefaultDockerClient implements DockerClient, Closeable {
       return connectTimeoutMillis;
     }
 
+    /**
+     * Set the timeout in milliseconds until a connection to Docker is established.
+     * A timeout value of zero is interpreted as an infinite timeout.
+     */
     public Builder connectTimeoutMillis(final long connectTimeoutMillis) {
       this.connectTimeoutMillis = connectTimeoutMillis;
       return this;
@@ -996,6 +1003,10 @@ public class DefaultDockerClient implements DockerClient, Closeable {
       return readTimeoutMillis;
     }
 
+    /**
+     * Set the SO_TIMEOUT in milliseconds. This is the maximum period of inactivity
+     * between receiving two consecutive data packets from Docker.
+     */
     public Builder readTimeoutMillis(final long readTimeoutMillis) {
       this.readTimeoutMillis = readTimeoutMillis;
       return this;
@@ -1005,6 +1016,9 @@ public class DefaultDockerClient implements DockerClient, Closeable {
       return dockerCertificates;
     }
 
+    /**
+     * Provide certificates to secure the connection to Docker.
+     */
     public Builder dockerCertificates(final DockerCertificates dockerCertificates) {
       this.dockerCertificates = dockerCertificates;
       return this;
@@ -1014,6 +1028,12 @@ public class DefaultDockerClient implements DockerClient, Closeable {
       return connectionPoolSize;
     }
 
+    /**
+     * Set the size of the connection pool for connections to Docker. Note that due to
+     * a known issue, DefaultDockerClient maintains two separate connection pools, each
+     * of which is capped at this size. Therefore, the maximum number of concurrent
+     * connections to Docker may be up to 2 * connectionPoolSize.
+     */
     public Builder connectionPoolSize(int connectionPoolSize) {
       this.connectionPoolSize = connectionPoolSize;
       return this;

